src/ucs/datastruct/arbiter.c

```diff
@@ -141,11 +141,17 @@ void ucs_arbiter_group_purge(ucs_arbiter_t *arbiter,
                 /* group became empty - deschedule it */
                 prev_group->list.next = &next_group->list;
                 next_group->list.prev = &prev_group->list;
+                if (arbiter->current == orig_head) {
+                    arbiter->current = next_group;
+                }
             } else if (orig_head != head) {
                 /* keep the group scheduled, but with new head element */
                 ucs_list_insert_replace(&prev_group->list,
                                         &next_group->list,
                                         &head->list);
+                if (arbiter->current == orig_head) {
```

Comment:
can this 'if' be moved outside if/else?

Reply:
we do not to change arbiter if the head is not changed. We can, but then we would need to add another alse branch with return statement.